### PR TITLE
Drop nose

### PR DESCRIPTION
**Pull request by nsoranzo**
It's deprecated and unmaintained, and breaks under Python 3.10 (xref. https://github.com/galaxyproject/planemo/pull/1234 ).

TODO:
- [x] lib/galaxy_test/base/instrument.py
- [x] lib/galaxy_test/driver/driver_util.py
- [ ] Fix converter tests

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/tool_shed/test/base/driver.py [hunk 1]
```diff
@@ -45,9 +38,9 @@
 
 
 class ToolShedTestDriver(driver_util.TestDriver):
-    """Instantial a Galaxy-style nose TestDriver for testing the tool shed."""
+    """Instantiate a Galaxy-style TestDriver for testing the tool shed."""
 
-    def setup(self):
+    def setup(self, config_object=None):
```
Comment (nsoranzo) [inline lib/tool_shed/test/base/driver.py]: Doesn't seem to be used?

```suggestion
    def setup(self):
```

File: test/functional/test_toolbox.py [hunk 2]
```diff
@@ -38,7 +37,6 @@ def do_it(self, tool_id=None, tool_version=None, test_index=0, resource_paramete
             resource_parameters=resource_parameters,
             test_index=test_index,
             tool_version=tool_version,
-            register_job_data=register_job_data,
```
Comment (jmchilton) [inline test/functional/test_toolbox.py:49]: This is needed by Planemo though right? Just making sure there is a plan for a new approach or does pytest accomplish capturing this data some other way?

Comment (mvdbeek) [reply test/functional/test_toolbox.py:49]: Only if we run tests via ./run_tests.sh, we could use `LocalManagedGalaxyEngine` in planemo, which would converge nicely with testing non-tool artefacts. I was wondering what your thoughts on that are.

Comment (nsoranzo) [reply test/functional/test_toolbox.py:49]: @jmchilton @mvdbeek What's the way forward here? Marius proposed on Matrix this patch for Planemo:

```diff
diff --git a/planemo/engine/test.py b/planemo/engine/test.py
index 0bfb4119..206dad37 100644
--- a/planemo/engine/test.py
+++ b/planemo/engine/test.py
@@ -17,18 +17,8 @@ def test_runnables(ctx, runnables, original_paths=None, **kwds):
     test_engine_testable = {RunnableType.galaxy_tool, RunnableType.galaxy_datamanager, RunnableType.directory}
     enable_test_engines = any(r.type not in test_engine_testable for r in runnables)
     enable_test_engines = enable_test_engines or engine_type != "galaxy"
-    if enable_test_engines:
-        ctx.vlog("Using test engine type %s" % engine_type)
-        with engine_context(ctx, **kwds) as engine:
-            test_data = engine.test(runnables)
-            ctx.vlog("engine.test returning [%s]" % test_data)
-            return_value = handle_reports_and_summary(ctx, test_data.structured_data, kwds=kwds)
-    else:
-        ctx.vlog("Running traditional Galaxy tool tests using run_tests.sh in Galaxy root %s" % engine_type)
-        kwds["for_tests"] = True
-        if kwds.get("update_test_data"):
-            non_copied_runnables = for_paths(original_paths)
-            kwds["test_data_target_dir"] = _find_test_data(non_copied_runnables, **kwds)
-        with galaxy_config(ctx, runnables, **kwds) as config:
-            return_value = run_in_config(ctx, config, **kwds)
-    return return_value
+    ctx.vlog("Using test engine type %s" % engine_type)
+    with engine_context(ctx, **kwds) as engine:
+        test_data = engine.test(runnables)
+        ctx.vlog("engine.test returning [%s]" % test_data)
+        return handle_reports_and_summary(ctx, test_data.structured_data, kwds=kwds)
```

Should we try to open a PR for planemo with this?

Comment (nsoranzo) [reply test/functional/test_toolbox.py:49]: After discussing this at the latest Backend WG meeting, I think the plan is:
- merge https://github.com/galaxyproject/planemo/pull/1232
- release galaxy-* packages for  22.01 on PyPI
- release new Planemo

Then the converter tests for this PR should pass, and we can merge it. Please correct me if I'm wrong.

**Comment by nsoranzo** [comment]
It looks like that rebasing on top of https://github.com/galaxyproject/galaxy/pull/13977 broke the Toolshed tests added to `lib/tool_shed/test/functional/` in that PR.
Also converters test broke after the rebase.
Help welcome!

**Comment by jmchilton** [comment]
Taking a quick look at that.

File: lib/tool_shed/test/base/twilltestcase.py [hunk 3]
```diff
@@ -11,6 +11,7 @@
     urlparse,
 )
 
+import pytest
```
Comment (nsoranzo) [inline lib/tool_shed/test/base/twilltestcase.py]:
```suggestion
```

**Comment by jmchilton** [comment]
I want to steal the code in here for running the tool shed via pytest... should I be patient and let this work out though? Is there something I can do to push this forward?

**Comment by nsoranzo** [comment]
> I want to steal the code in here for running the tool shed via pytest... should I be patient and let this work out though? Is there something I can do to push this forward?

I think if you could help getting https://github.com/galaxyproject/planemo/pull/1232 merged, then this one should quickly become ready as well.

**Comment by nsoranzo** [comment]
After rebase, the converters tests pass (Selenium failure unrelated), so ready to merge!

File: lib/galaxy/tool_util/linters/tests.py [hunk 4]
```diff
@@ -9,8 +9,7 @@
 from ..verify import asserts
 
 
-# Misspelled so as not be picked up by nosetests.
-def lint_tsts(tool_xml, lint_ctx):
+def lint_tests(tool_xml, lint_ctx):
```
Comment (mvdbeek) [inline lib/galaxy/tool_util/linters/tests.py:12]: Could this be in use with people using tool-util as a library ?

Comment (nsoranzo) [reply lib/galaxy/tool_util/linters/tests.py:12]: Not by planemo, ephemeris or cwltool. Any other I should check?

**Review by mvdbeek — APPROVED** [review]

**Pull request merged**